src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -1176,7 +1176,7 @@ private EncodedKey FindPageFor(ReadOnlySpan<byte> key, ref IteratorCursorState c
             return encodedKey;
         }
 
-        private EncodedKey SearchPageAndPushNext(in EncodedKey key, ref IteratorCursorState cstate)
+        private EncodedKey SearchPageAndPushNext(EncodedKey key, ref IteratorCursorState cstate)
```

Comment:
Doesn't this force a copy, vs. a pointer ref?

Reply:
For the record, I believe the issues that it is complaining about is that you have a call to an `in` (which is *by-ref*)  inside a call that receives the parameter as 'in` which is *by-ref*. The compiler is enforcing that the value is actually *in the stack* and that is why it fails. There must be a correctness issues somewhere.